src/psyclone/psyir/nodes/node.py

```diff
@@ -1072,36 +1072,45 @@ def sameParent(self, node_2):
             return False
         return self.parent is node_2.parent
 
-    def walk(self, my_type, stop_type=None):
+    def walk(self, my_type, stop_type=None, depth=None):
         ''' Recurse through the PSyIR tree and return all objects that are
         an instance of 'my_type', which is either a single class or a tuple
         of classes. In the latter case all nodes are returned that are
         instances of any classes in the tuple. The recursion into the tree
         is stopped if an instance of 'stop_type' (which is either a single
         class or a tuple of classes) is found. This can be used to avoid
         analysing e.g. inlined kernels, or as performance optimisation to
-        reduce the number of recursive calls.
+        reduce the number of recursive calls. The recursion into the tree is
+        also stopped if the (optional) 'depth' level is reached.
 
         :param my_type: the class(es) for which the instances are collected.
         :type my_type: type | Tuple[type, ...]
         :param stop_type: class(es) at which recursion is halted (optional).
         :type stop_type: Optional[type | Tuple[type, ...]]
+        :param depth: the depth value the instances must have (optional).
+        :type depth: Optional[int]
 
         :returns: list with all nodes that are instances of my_type \
                   starting at and including this node.
         :rtype: List[:py:class:`psyclone.psyir.nodes.Node`]
 
         '''
         local_list = []
-        if isinstance(self, my_type):
+        if isinstance(self, my_type) and (depth is None or self.depth == depth):
```

Comment:
pylint says this line is too long (and would `depth in [None, self.depth]` work?)

Reply:
Applied this suggestion in 9d96edf37.